sympy/integrals/transforms.py

```diff
@@ -1785,6 +1785,34 @@ def _laplace_rule_diff(f, t, s, doit=True, **hints):
     return False, (fn, p, c)
 
 
+def _laplace_rule_sdiff(f, t, s, doit=True, **hints):
+    """
+    """
+    fn, p, c = _laplace_ct(f)
+
+    if fn.is_Mul:
+        pfac = [fac for fac in Mul.make_args(fn) if fac.is_polynomial(t)]
+        if len(pfac)>0:
+            pex = prod(pfac)
+            pc = Poly(pex, t).all_coeffs()
+            N = len(pc)
+            if N>1:
+                debug('_laplace_apply_rules match:')
+                debug('      f, n: %s, %s'%(fn, pfac))
+                debug('      rule: frequency derivative')
+                rem = (fn/pex).simplify(doit=False)
```

Comment:
I think it would be better to just collect the factors that have `is_polynomial = False` rather than dividing and calling simplify.

Reply:
Good idea, thanks.